meta/data.go

```diff
@@ -622,6 +622,11 @@ func (sgi *ShardGroupInfo) Contains(timestamp time.Time) bool {
 	return !sgi.StartTime.After(timestamp) && sgi.EndTime.After(timestamp)
 }
 
+// Overlaps return whether the shard group contains data for the time range between min and max
+func (sgi *ShardGroupInfo) Overlaps(min, max time.Time) bool {
+	return !sgi.StartTime.After(max) && sgi.EndTime.After(min)
+}
```

Comment:
I don't think this works.  Based on this, and the start/end times of a shard always falling on the exact hour, you can match more than one shard for the same point:

[play example](http://play.golang.org/p/r0EWEYMt4S)